### DISCUSSION TITLE
%matplotlib widget legend outside of plot gets cut off

### DISCUSSION THREAD

**Discussion by rcontinuum** [microsoft/vscode-jupyter#10470] [Questions and Answers] [answered]
Hello, 
the following plot code 
```
import numpy as np
import matplotlib.pyplot as plt
%matplotlib widget

x = np.linspace(0, 20, 200)
plt.plot(x, np.sin(x), label="sine_long_label")
plt.legend(bbox_to_anchor =(1.1, 1.0))
plt.show()
```
cuts off almost the complete legend. See the red arrow for the small visible part of the legend.
Remark: Without the line `%matplotlib widget` the legend is fully visible. 
![Figure1](https://user-images.githubusercontent.com/93375447/174090100-317296c4-f4a4-4cd3-b9d3-ad3a6959f521.png)

Any hints?

**Comment by IanMatthewHuff** [comment]
Hmm, seems like we might not be getting the initial sizing right here? I just tried it out myself and I note that there is a small grey resizing arrow in the bottom right. 
<img width="261" alt="image" src="https://user-images.githubusercontent.com/812783/174159255-2c8e55a9-5719-4891-8a2b-77ffea67c5b8.png">
Using that, I was able to drag out more space to see the legend, but that doesn't seem optimal. Might work as a workaround though:
<img width="1086" alt="image" src="https://user-images.githubusercontent.com/812783/174159393-99d8fef7-2724-4e1a-9e7f-55fb31afd206.png">

**Reply by rcontinuum** [reply]
Thanks!
I can conform that by dragging the grey triangle and making the plot much larger in x-direction at a certain width the legend "pops in". I had to drag it in x-direction to about 2 to 3 times the initial width. 
Shall I open an Issue concerning this? Just as a reminder to fix this behaviour at a later point. 

Cheers

**Comment by DonJayamanne** [comment, marked as answer]
I've tried this myself in classic Jupyter Notebook and the same problem exists there, hence this is not specific to VS Code, ie not a problem caused by VS Code. Meaning its not something we can necessarily fix.

Here's a screenshot of the exact same issue in Jupyter Notebooks.

<img width="755" alt="Screen Shot 2022-06-17 at 18 10 29" src="https://user-images.githubusercontent.com/1948812/174256203-7a6beebe-1a7d-4882-b36f-cfc17f9f2263.png">

**Comment by rcontinuum** [comment]
I see, there is something off in Jupyter. 
I just want to report that I see the same effect (legend outside of plot gets cut off ) also with using
external plot windows like `%matplotlib qt5` (or even `%matplotlib wx`)  instead of `%matplotlib widget`. 
So, I close this Discussion here.

Thank you.

**Comment by Josverl** [comment]
FWIW , The resize handle (rey - right-bottom corner) can be removed/disabled using 
```
# Disable the resizing feature
fig.canvas.resizable = False
```
